hera_cal/utils.py

```diff
@@ -263,7 +326,233 @@ def combine_calfits(files, fname, outdir=None, overwrite=False, broadcast_flags=
     uvc.write_calfits(output_fname, clobber=True)
 
 
+def get_miriad_times(filepaths, add_int_buffer=False):
+    """
+    Use aipy to get filetimes for Miriad file paths.
+    All times mark the center of an integration, which is not the standard Miriad format.
+
+    Parameters:
+    ------------
+    filepaths : type=list, list of filepaths
+
+    add_int_buffer : type=bool, if True, extend start and stop times by a single integration duration 
+        except for start of first file, and stop of last file.
+
+    Output: (file_starts, file_stops, int_times)
+    -------
+    file_starts : file starting point in LST [radians]
+
+    file_stops : file ending point in LST [radians]
+
+    int_times : integration duration in LST [radians]
+    """
+    _array = True
+    # check filepaths type
+    if type(filepaths) == str:
+        _array = False
+        filepaths = [filepaths]
+
+    # form empty lists
+    file_starts = []
+    file_stops = []
+    int_times = []
+
+    # get Nfiles
+    Nfiles = len(filepaths)
+
+    # iterate over filepaths and extract time info
+    for i, f in enumerate(filepaths):
+        uv = aipy.miriad.UV(f)
+        # get integration time
+        int_time = uv['inttime'] * 2*np.pi / (23.9344699*3600.)
+        # get start and stop
+        start = uv['lst']
+        stop = start + (uv['ntimes']-1) * int_time
+        # add integration buffer to beginning and end if desired
+        if add_int_buffer:
+            if i != 0:
+                start -= int_time
+            if i != (Nfiles-1):
+                stop += int_time
+        # add half an integration to get center of integration
+        start += int_time / 2
+        stop += int_time / 2
+        file_starts.append(start)
+        file_stops.append(stop)
+        int_times.append(int_time)
+
+    file_starts = np.array(file_starts)
+    file_stops = np.array(file_stops)
+    int_times = np.array(int_times)
+
+    # make sure times don't wrap
+    file_starts[np.where(file_starts < 0)] += 2*np.pi
+    file_stops[np.where(file_stops >= 2*np.pi)] -= 2*np.pi
+
+    if _array is False:
+        file_starts = file_starts[0]
+        file_stops = file_stops[0]
+        int_times = int_times[0]
+
+    return file_starts, file_stops, int_times
+
+
+def data_to_miriad(fname, data, lst_array, freq_array, antpos, time_array=None, flags=None,
```

Comment:
Does this do everything we need for omnical's vis.uvfits and for omni_apply's .uvcO and xrfi_apply's .uvcOR files? If so, perhaps you want to make an issue that we should be updating those functions to use this. Otherwise we're duplicating code and it makes it harder to keep up with changes to pyuvdata.

Reply:
I think they are doing fundamentally different things. For `omni_apply`, there is a strong case to be made that we *should* be reading-in a miriad file and using that as a UVData template which we then edit and write out. In this case, we are creating a UVData object from scratch. So I'm not sure if `data_to_miriad` can replace what's in `omni.py`, but maybe...

Reply:
I feel like our interfaces to pyuvdata are a complete mess right now. We've got lots of duplicated functionality and a lot of hard-coded code is all over the place. I'm going to make a new issue.